From 88f5dc7f382d729f18790c52f0ab543ae46177a0 Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Tue, 10 Mar 2026 12:47:17 -0700
Subject: [PATCH 6/6] drop .so

---
 CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 162463f..85f6ce0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,7 +91,6 @@ file(GLOB SOURCES "Root/*.cc" )
 file(GLOB LIB_HEADERS "FastFrames/*.h" )
 
 add_library( FastFrames SHARED ${LIB_HEADERS} ${SOURCES} )
-set_target_properties(FastFrames PROPERTIES SUFFIX ".so")
 target_include_directories(FastFrames
   PUBLIC
     $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
@@ -118,14 +117,14 @@ endif()
 
 # Build the Python module for C++ logger
 add_library( cppLogger MODULE python_wrapper/utils/Logger.cxx )
-set_target_properties(cppLogger PROPERTIES SUFFIX ".so" PREFIX "")
+set_target_properties(cppLogger PROPERTIES PREFIX "")
 target_include_directories( cppLogger PRIVATE ${Python3_INCLUDE_DIRS} )
 set_target_properties( cppLogger PROPERTIES
    OUTPUT_NAME "cppLogger" )
 
 # Build the Python module for config reader
 add_library( ConfigReaderCpp MODULE python_wrapper/utils/ConfigReaderCpp.cxx ${LIB_HEADERS} )
-set_target_properties(ConfigReaderCpp PROPERTIES SUFFIX ".so" PREFIX "")
+set_target_properties(ConfigReaderCpp PROPERTIES PREFIX "")
 target_include_directories( ConfigReaderCpp PRIVATE ${Python3_INCLUDE_DIRS} )
 target_link_libraries( ConfigReaderCpp PRIVATE ${Boost_LIBRARIES} FastFrames )
 set_target_properties( ConfigReaderCpp PROPERTIES
-- 
2.45.2

